### DISCUSSION TITLE
How can I simulate a transmitter with multiple array panels?

### DISCUSSION THREAD

**Discussion by wayne-hanwei** [NVlabs/sionna#127] [Q&A] [answered]
How can I simulate a transmitter with multiple array panels? 
Say for example, one transmiter equipped with two planar array panels. For each panel, it is a 4x8 planar array. The inter-distance between these two panel is about 1~2 meters, along horizontal axis.

**Comment by faycalaa** [comment]
Hi @wayne-hanwei ,

The documentation of the channel API shows how to create a `PanelArray`:

https://nvlabs.github.io/sionna/api/channel.wireless.html#sionna.channel.tr38901.PanelArray

The panel spacing and antenna element spacing can be configured. The only constraint is that the panels must have the same geometry.

Here is a code snippet showing how to create two 4x8 array panels:

```
array = PanelArray(num_rows_per_panel = 4,
                   num_cols_per_panel = 8,
                   polarization = 'single',
                   polarization_type = 'V',
                   antenna_pattern = '38.901',
                   carrier_frequency = 3.5e9,
                   num_cols = 2,
                   panel_horizontal_spacing = 20) # Horizontal spacing of 20 wavelengths between panel centers
```

**Reply by wayne-hanwei** [reply]
Thanks! I will try and get back soon : )

**Reply by wayne-hanwei** [reply]
Hi @faycalaa ,

It seems that the PanelArray module only support statistical channel models (e.g., 38.901). Is it possible to create a multi-panel antenna array in the raytracing channel model?

Actually, my goal is to evaluate the behavior of a multi-panel array, or distributed MIMO in general, in a specific radio propagation environment. What is most interesting to me is to find out the optimal orientation of each panel, in order to adapt to the environment and the locations of users.

One alternative way, I believe, is to create multiple transmitters with close inter-distance, since the orientation of each transmitter is freely trainable. However, it create another problem of joint processing of multiple transmitters, which seems to be a lot of work to be done.

Any idea on this? Glad to hear about it.

**Reply by faycalaa** [reply, marked as answer]
What you have suggested should work: Setting the scene `tx_array` to configure the geometry of a panel and instantiating different transmitters to simulate multiple panels each  with its own orientation.
